### PR TITLE
ci: fix benchmarks cleanup

### PR DESCRIPTION
**Pull request by reakaleek**
## Motivation/summary

The benchmarks workflow does not clean up all the resources it's creating.

In this [workflow run](https://github.com/elastic/apm-server/actions/runs/11183981002) you can see that the [Spin up benchmark environment](https://github.com/elastic/apm-server/actions/runs/11183981002/job/31093763569#step:15:729) step creates 22 resources.

But the [Teardown benchmark environment](https://github.com/elastic/apm-server/actions/runs/11183981002/job/31093763569#step:30:381) only deletes 9 resources.

The reason for it is the differently passed `TF_VAR_*` environment variables in both steps.

This change fixes it and aligns the passed tf vars to both apply and destroy.

## Checklist

- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [ ] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

I was able to test this change in https://github.com/elastic/apm-server/actions/runs/11277362803/job/31363323001, where you can find 22 resources being destroyed.

### CODE REVIEW

File: .github/workflows/benchmarks.yml [hunk 1]
```diff
@@ -133,10 +134,6 @@ jobs:
           admin_console_url=$(terraform output -raw admin_console_url)
           echo "admin_console_url=$admin_console_url" >> "$GITHUB_OUTPUT"
           echo "-> infra setup done"
-        env:
-          TF_VAR_worker_region: ${{ env.AWS_REGION }}
```
Comment (reakaleek) [inline .github/workflows/benchmarks.yml:137]: `env.AWS_REGION` doesn't exist. Hence I removed it.

**Review by 1pkg — APPROVED** [review]

File: .github/workflows/benchmarks.yml (hunk 1 above)
Comment (v1v) [reply .github/workflows/benchmarks.yml:137]: can you add the link to the reference that included it? In case there are some rationale?

**Review by v1v — APPROVED** [review]

File: .github/workflows/benchmarks.yml (hunk 1 above)
Comment (reakaleek) [reply .github/workflows/benchmarks.yml:137]: https://github.com/elastic/apm-server/pull/13884/files#diff-22eef62c9d7b36d02f5fd5aaada92702e22d6795c464239cb7b8fe0f26ea1e1cR137

Comment (reakaleek) [reply .github/workflows/benchmarks.yml:137]: is this what you meant?

**Pull request merged**